datafusion/expr/src/logical_plan/plan.rs

```diff
@@ -293,48 +293,120 @@ pub enum LogicalPlan {
     DelimGet(DelimGet),
 }
 
+#[derive(Clone, Debug, Eq, PartialOrd, Hash)]
+pub struct CorrelatedColumnInfo {
+    pub col: Column,
+    pub data_type: DataType,
+    pub depth: usize,
+}
+
+impl PartialEq for CorrelatedColumnInfo {
+    fn eq(&self, other: &Self) -> bool {
+        self.col == other.col && self.data_type == other.data_type
+    }
+}
+
 #[derive(Debug, Clone, Eq)]
 pub struct DelimGet {
+    // TODO: is it necessary to alias?
+    pub table_name: TableReference,
+    pub columns: Vec<Column>,
     /// The schema description of the output
     pub projected_schema: DFSchemaRef,
-    pub columns: Vec<Column>,
-    pub table_index: usize,
-    pub delim_types: Vec<DataType>,
+    // TODO: add more variables as needed.
 }
 
 impl DelimGet {
-    pub fn try_new(
-        table_index: usize,
-        columns: Vec<Column>,
-        delim_types: &[DataType],
-        projected_schema: DFSchemaRef,
-    ) -> Self {
-        Self {
-            columns,
-            projected_schema,
-            table_index,
-            delim_types: delim_types.to_owned(),
+    pub fn try_new(correlated_columns: &Vec<CorrelatedColumnInfo>) -> Result<Self> {
+        if correlated_columns.is_empty() {
+            // return plan_err!("failed to construct DelimGet: empty correlated columns");
+            // TODO: revisit if dummy dependent join is nesessary.
+            return Ok(Self {
+                table_name: TableReference::bare("empty scan"),
+                columns: vec![],
+                projected_schema: Arc::new(DFSchema::empty()),
+            });
+        }
+
+        let correlated_columns: Vec<CorrelatedColumnInfo> = correlated_columns
+            .into_iter()
+            .map(|info| {
+                // Add "_d" suffix to the relation name
+                let col = if let Some(ref relation) = info.col.relation {
+                    let new_relation =
+                        Some(TableReference::bare(format!("{}_d", relation)));
+                    Column::new(new_relation, info.col.name.clone())
+                } else {
+                    info.col.clone()
+                };
+
+                CorrelatedColumnInfo {
+                    col,
+                    data_type: info.data_type.clone(),
+                    depth: info.depth,
+                }
+            })
+            .collect();
+
+        // Extract the first table reference to validate all columns come from the same table
+        let first_table_ref = correlated_columns[0].col.relation.clone();
```

Comment:
It is possible for a delim get to correspond with multiple tables, example is here:
https://github.com/duongcongtoai/arrow-datafusion/blob/e2b19981ee610d40d0af797a8385eed4d4e22651/datafusion/optimizer/src/decorrelate_dependent_join.rs#L1178

I also tested with duckdb using this query
```
D explain select * from t1 where exists(select * from t2 where t2.id2 =1 and  exists(select * from t3 where id3=id1 and val3=val2));
```

In the final level there will be a delim_get to scan for unique values of (t1.id1 and t2.val2)